nextcloud/server
# fix(encryption): Clicking default module in UI sets bogus value

* Resolves: #44532 
* Likely fixes #43123 too

## Summary

`update()` was hard coded to return a yes/no string value just to handle `enableEncryption()`'s value. Since `update()` is also used for setting the default encryption module (which is not a simple yes/no value), whenever the default encryption module radio button was clicked the server value was being set to `yes` too (which is bogus).

https://github.com/nextcloud/server/blob/497a8631230109f809e23f117ec7910d8fb63579/apps/settings/src/components/Encryption.vue#L132-L156

https://github.com/nextcloud/server/blob/497a8631230109f809e23f117ec7910d8fb63579/apps/settings/src/components/Encryption.vue#L60-L68

### How to test:

- Enable Encryption app under *Apps*
- Toggle on Server-side encryption under *Admin Settings->Security*
- Click *Enable encryption*
- Click anywhere on the *Default encryption module* radio button area
- Check the server-side value of `default_encryption_module` (e.g. `occ config:list core`)

Correct value: `OC_DEFAULT_MODULE` (which it'll be if you're careful not to click anywhere initially)
Bogus value: `yes` (which it'll change to if you click anywhere without this PR)

If your default_encryption_module is set to `yes` try this command to manually remedy:

`occ encryption:set-default-module OC_DEFAULT_MODULE`

## TODO


## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
